plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,489 @@
+  _MERGING_BYTES = b'merging'
+  # DEFAULT_REDIS_URL is public so that it appears in generated documentation.
+  DEFAULT_REDIS_URL = 'redis://127.0.0.1/0'
+
+  def __init__(
+      self, storage_type=definitions.STORAGE_TYPE_TASK,
+      session_identifier=None, task_identifier=None):
+    """Initializes a Redis store.
+
+    Args:
+      storage_type (Optional[str]): storage type.
+      session_identifier (Optional[str]): session identifier, formatted as
+          a UUID.
+      task_identifier (Optional[str]): unique identifier of the task the store
+          will store containers for. If not specified, an identifier will be
+          generated.
+
+    Raises:
+      ValueError: if the storage type is not supported.
+    """
+    if storage_type != definitions.STORAGE_TYPE_TASK:
+      raise ValueError('Unsupported storage type: {0:s}.'.format(
+          storage_type))
+    super(RedisStore, self).__init__()
+    if not session_identifier:
+      session_identifier = str(uuid.uuid4())
+    self._session_identifier = session_identifier
+    if not task_identifier:
+      task_identifier = str(uuid.uuid4())
+    self._task_identifier = task_identifier
+    self._redis_client = None
+    self.serialization_format = definitions.SERIALIZER_FORMAT_JSON
+
+  @classmethod
+  def _SetClientName(cls, redis_client, name):
+    """Attempts to sets a Redis client name.
+
+    This method ignores errors from the Redis server or exceptions
+    indicating the method is missing, as setting the name is not a critical
+    function, and it is not currently supported by the fakeredis test library.
+
+    Args:
+      redis_client (Redis): an open Redis client.
+      name (str): name to set.
+    """
+    try:
+      redis_client.client_setname(name)
+    except redis.ResponseError as exception:
+      logger.debug(
+          'Unable to set redis client name: {0:s} with error: {1!s}'.format(
+              name, exception))
+
+  def _AddAttributeContainer(
+      self, container_type, container, serialized_data=None):
+    """Adds an attribute container to the store.
+
+   Args:
+      container_type (Type): container type attribute of the container being
+          added.
+      container (AttributeContainer): unserialized attribute container.
+      serialized_data (Optional[bytes]): serialized form of the container.
+    """
+    self._RaiseIfNotWritable()
+
+    identifier = identifiers.RedisKeyIdentifier()
+    container.SetIdentifier(identifier)
+
+    if not serialized_data:
+      serialized_data = self._SerializeAttributeContainer(container)
+
+    container_key = self._GenerateRedisKey(container_type)
+    string_identifier = identifier.CopyToString()
+    self._redis_client.hset(container_key, string_identifier, serialized_data)
+
+  def _GenerateRedisKey(self, subkey):
+    """Generates a Redis key inside the appropriate namespace.
+
+    Args:
+      subkey (str): Redis key to be prefixed with the namespace value.
```

Comment:
Maybe instead of subkey use name_suffix or namespace since that seems what this value is used for.

Reply:
Renamed to key_suffix